### PR TITLE
Error handling enhancement

### LINKED ISSUE #7: Able to access error message from error object
- Not able to get the correct reason of error while creating token failed.
- The error messages are not localized.

### PR DESCRIPTION
**Pull request by vc7**
## Issue

Resolve #7.

## Check Items.

- [ ] Able to get error messages from `error.localizedDescription`
- [ ] Able to set prefer locale to `PAYAPIClient`

### CODE REVIEW

File: Sources/PAYError.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    
+    // MARK: - Decodable
+    
+    init(_ e: Extractor) {
+        status = (try? e <| "status") ?? 0
+        message = try? e <| "message"
+        param = try? e <| "param"
+        code = try? e <| "code"
+        type = try? e <| "type"
+    }
```
Comment (natsuki-yamanaka) [inline Sources/PAYError.swift]: Why don't you add description like this?

`status: 402 message: Invalid card number param: card[number] code: invalid_number type: card_error`

when `print(payError)` exec.

If that's okay, just add like below. 

```
override public var description: String { return "status: \(status ?? "") message: \(message ?? "") param: \(param ?? "") code: \(code ?? "") type: \(type ?? "")" }
```

File: Sources/APIError.swift [hunk 2]
```diff
@@ -9,11 +9,76 @@
...
+            userInfo[PAYErrorServiceErrorObject] = error
+            return NSError(domain: PAYErrorDomain,
+                           code: PAYErrorServiceError,
+                           userInfo: userInfo)
+        case .invalidJSON(let json):
+            userInfo[PAYErrorInvalidJSONObject] = json
+            return NSError(domain: PAYErrorDomain,
+                           code: PAYErrorInvalidJSON,
+                           userInfo: userInfo)
+        }
```
Comment (natsuki-yamanaka) [inline Sources/APIError.swift]: This class would be needed workaround if someone want to PayError instance when thrown APIError.
like this
```
        //in example-swift showError(error: Error) method
        if let error = error as? APIError {
            if let payError = error.nsErrorValue().userInfo[PAYErrorServiceErrorObject] as? PAYError {
                NSLog("payError \(payError)")
            }
        }
```

How about add method below.
```
    public func payErrorValue()-> PAYErrorType? {
        switch self {
        case .serviceError(let error):
            return error
        default:
            return nil
        }
    }
```

**Review by natsuki-yamanaka — CHANGES_REQUESTED** [review]
Sorry, I'd like you to check it.

File: Sources/PAYError.swift (hunk 1 above)
Comment (vc7) [reply Sources/PAYError.swift]: sounds great!

File: Sources/APIError.swift (hunk 2 above)
Comment (vc7) [reply Sources/APIError.swift]: For easy accessing belonging PAYError object, it's is a good idea!
I'll add the helper method on.

File: README.md [hunk 3]
```diff
@@ -19,8 +19,8 @@
 ## サンプルコード
 
 - Apple Pay: https://github.com/payjp/apple-pay-example
-- CreditCard (Swift): https://github.com/payjp/payjp-ios/tree/master/example-swift
-- CreditCard (Objective-C): https://github.com/payjp/payjp-ios/tree/master/example-objc
+- CreditCard (Swift, Carthage): https://github.com/payjp/payjp-ios/tree/master/example-swift
+- CreditCard (Objective-C, CocoaPods): https://github.com/payjp/payjp-ios/tree/master/example-objc
 
 ## 動作環境
```
Comment (natsuki-yamanaka) [inline README.md:25]: I want you to modified some line like below.
` - Xcode 9.0以上`
↓
` - 最新安定版のXcode`

Comment (vc7) [reply README.md:25]: 👍

**Review by natsuki-yamanaka — APPROVED** [review]

**Pull request merged**